app/models/exercise_with_solution.rb

```diff
@@ -3,9 +3,8 @@ class ExerciseWithSolution < SimpleDelegator
 
   delegate :approved?, :in_progress?, to: :solution
 
-  def initialize(exercise, solution)
+  def initialize(exercise)
```

Comment:
I think this should still take two arguments. If the class ever get reused the `exercise.solutions.first` thing would be unexpected behaviour.

Reply:
Thanks! I agree that's a better solution.